subsys/pcd/src/pcd.c

```diff
@@ -183,10 +208,11 @@ static int network_core_pcd_cmdset(const void *src_addr, size_t len, bool wait)
 		k_busy_wait(1 * USEC_PER_SEC);
 
 		err = pcd_fw_copy_status_get();
-	} while (err == PCD_STATUS_COPY);
+	} while (err == command_status);
 
-	if (err == PCD_STATUS_COPY_FAILED) {
+	if (err == PCD_STATUS_FAILED) {
 		LOG_ERR("Network core update failed");
+		network_core_pcd_tidy();
```

Comment:
I'm a little bit uncertain if this will break things. It resets the network core back to non-secure if either the network core update fails or the version number reading failed.

Reply:
But why do we need to add it now?

Reply:
In my mind: "Because you could end up reading the version and it failed, try to start the application. This would fail as the network core would be in secure while the application expects non-secure." this was the logic. I'll go through the code once more to see if this is true.